modules/objdetect/include/opencv2/objdetect/barcode.hpp

```diff
@@ -57,6 +57,27 @@ class CV_EXPORTS_W_SIMPLE BarcodeDetector : public cv::GraphicalCodeDetector
                                       CV_OUT std::vector<std::string> &decoded_info,
                                       CV_OUT std::vector<std::string> &decoded_type,
                                       OutputArray points = noArray()) const;
+
+    /// downsample image if minimum dimension is larger than threshold (default 512)
+    CV_PROP_RW float detectorThreshDownSamplingLimit;
```

Comment:
In public API only "*Parameters" classes are allowed to have properly fields.
Algorithms classes should have setters/getters in public API and properly fields in internal implementations.